src/lib/pipeline.js

```diff
@@ -17,9 +17,8 @@ class PipeLine {
           console.log(`Error running data through pipe ${pipeSection.constructor.name} \n ${error}`);
         }
       }
-      if (normalisedEvents) {
-        await this.pipeOutputCb( this.rawData.id, normalisedEvents,);
-      }
+
```

Comment:
I would have put a comment here

// We always want to call pipeOutputCb even if normalisedEvents is empty so pipeOutputCb will record that we have checked this bit of data

Just stops someone trying to add it back later without realising there is a good reason

But maybe later